src/pydvl/value/shapley/classwise.py

```diff
@@ -0,0 +1,301 @@
+    :param eps: The threshold when updating using the truncated monte carlo estimator.
+    :param n_jobs: Number of jobs to run in parallel.
+    :param config: Parallel configuration.
+    :return: ValuationResult object with the data values.
+    """
+
+    map_reduce_job: MapReduceJob[NDArray, ValuationResult] = MapReduceJob(
+        u.data.indices,
+        map_func=_class_wise_shapley_worker,
+        reduce_func=lambda results: reduce(operator.add, results),
+        map_kwargs=dict(u=u, done=done, progress=progress, eps=eps),
+        n_jobs=n_jobs,
+        config=config,
+    )
+    result = map_reduce_job()
+    y_train = u.data.y_train
+    unique_labels = np.unique(np.concatenate((y_train, u.data.y_test)))
+
+    if normalize_score:
+
+        for idx_label, label in enumerate(unique_labels):
+
+            u.scorer.label = label
+            active_elements = y_train == label
+            label_set = np.where(active_elements)[0]
+
+            u.model.fit(u.data.x_train, u.data.y_train)
+            in_cls_acc, _ = _estimate_in_out_cls_accuracy(
+                u.model, u.data.x_test, u.data.y_test, label
+            )
+
+            sigma = np.sum(result.values[label_set])
+            result.values[label_set] *= in_cls_acc / sigma
+
+    return result
+
+
+def __class_complement_conditional_sampler(
+    u: Utility,
+    label: int,
+    result: ValuationResult,
+    *,
+    active_indices: IntArray = None,
+    eps: float = 1e-4,
+    num_samples: int = 10,
+) -> ValuationResult:
+    """
+    Samples a random subset of the complement set and computes the truncated monte carlo estimator.
+
+    :param u: Utility object with model, data, and scoring function. The scoring function has to be of type CSScorer.
+    :param label: The label for which the complement set shall be sampled.
+    :param result: The current result object.
+    :param active_indices: The indices of the active elements.
+    :param eps: The threshold when updating using the truncated monte carlo estimator.
+    :param num_samples: The number of subset samples to be drawn from the complement set.
+    :return: The updated result object.
+    """
+    if active_indices is None:
+        active_indices = u.data.indices
+
+    _, y_train = u.data.get_training_data(active_indices)
+    label_set, complement_label_set = __split_into_label_sets(
+        y_train, label, active_indices
+    )
+    _, complement_y_train = u.data.get_training_data(complement_label_set)
+
+    for subset_complement in random_powerset_group_conditional(
+        complement_label_set,
+        complement_y_train,
+        n_samples=num_samples,
+    ):
+        result = __truncated_permutation_mcmc_sampler(
+            u, np.random.permutation(label_set), subset_complement, result, atol=eps
+        )
+
+    return result
+
+
+def _estimate_in_out_cls_accuracy(
+    model: SupervisedModel, x: np.ndarray, labels: np.ndarray, label: np.int_
```

Comment:
Use `NDArray[type]` for the types. Also, since you name the data `x`, I think you should name the labels `y`, for consistency